pkg/operation/botanist/controlplane/resourcemanager/resource_manager.go

```diff
@@ -0,0 +1,553 @@
+		cmd = append(cmd, fmt.Sprintf("--leader-election-lease-duration=%s", *r.config.LeaseDuration))
+	}
+	cmd = append(cmd, fmt.Sprintf("--leader-election-namespace=%s", r.namespace))
+	if r.config.RenewDeadline != nil {
+		cmd = append(cmd, fmt.Sprintf("--leader-election-renew-deadline=%s", *r.config.RenewDeadline))
+	}
+	if r.config.RetryPeriod != nil {
+		cmd = append(cmd, fmt.Sprintf("--leader-election-retry-period=%s", *r.config.RetryPeriod))
+	}
+	if r.config.ConcurrentSyncs != nil {
+		cmd = append(cmd, fmt.Sprintf("--max-concurrent-workers=%v", *r.config.ConcurrentSyncs))
+	}
+	cmd = append(cmd, fmt.Sprintf("--metrics-bind-address=:%v", metricsPort))
+	if r.config.WatchedNamespace != nil {
+		cmd = append(cmd, fmt.Sprintf("--namespace=%s", *r.config.WatchedNamespace))
+	}
+	if r.config.ResourceClass != nil {
+		cmd = append(cmd, fmt.Sprintf("--resource-class=%s", *r.config.ResourceClass))
+	}
+	if r.config.SyncPeriod != nil {
+		cmd = append(cmd, fmt.Sprintf("--sync-period=%s", *r.config.SyncPeriod))
+	}
+	if r.config.TargetDisableCache != nil {
+		cmd = append(cmd, "--target-disable-cache")
+	}
+	if r.config.KubeConfig != nil {
+		cmd = append(cmd, fmt.Sprintf("--target-kubeconfig=%s/%s", volumeMountPathKubeconfig, secrets.DataKeyKubeconfig))
+	}
+	return cmd
+}
+
+func (r *ResourceManager) ensureServiceAccount(ctx context.Context) error {
+	serviceAccount := r.emptyServiceAccount()
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, serviceAccount, func() error {
+		serviceAccount.Labels = *r.config.DefaultLabels
+		return nil
+	})
+	return err
+}
+
+func (r *ResourceManager) emptyServiceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: serviceAccountName, Namespace: r.namespace}}
+}
+
+func (r *ResourceManager) ensureVPA(ctx context.Context) error {
+	vpa := r.emptyVPA()
+	vpaUpdateMode := autoscalingv1beta2.UpdateModeAuto
+
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, vpa, func() error {
+		vpa.Labels = *r.config.DefaultLabels
+		vpa.Spec.TargetRef = &autoscalingv1.CrossVersionObjectReference{
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+			Name:       v1beta1constants.DeploymentNameGardenerResourceManager,
+		}
+		vpa.Spec.UpdatePolicy = &autoscalingv1beta2.PodUpdatePolicy{
+			UpdateMode: &vpaUpdateMode,
+		}
+		return nil
+	})
+	return err
+}
+
+func (r *ResourceManager) emptyVPA() *autoscalingv1beta2.VerticalPodAutoscaler {
+	return &autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "gardener-resource-manager", Namespace: r.namespace}}
+}
+
+func appLabel() *map[string]string {
+	return &map[string]string{
+		v1beta1constants.LabelApp: "gardener-resource-manager",
+	}
+}
+
+// SetKubeConfig enables the ResourceManager to deploy resources into a different
+// cluster than the one it is running in.
+func (r *ResourceManager) SetKubeConfig(k *component.Secret) { r.config.KubeConfig = k }
```

Comment:
Can we call this `SetSecrets()` like in other components? Yes, today there is only the kubeconfig secret, but maybe this will change in the future. Though, main motivation here would be similarity to other components.